pygaggle/reader/dpr_reader.py

```diff
@@ -56,38 +60,49 @@ def get_tokenizer(
         return DPRReaderTokenizer.from_pretrained(pretrained_tokenizer_name_or_path)
 
     def predict(self, query: Query, texts: List[Text]) -> List[Answer]:
-        answers = []
-        for i in range(0, len(texts), self.batch_size):
-            encoded_inputs = self.tokenizer(
-                questions=query.text,
-                titles=list(map(lambda t: t.title, texts[i: i+self.batch_size])),
-                texts=list(map(lambda t: t.text, texts[i: i+self.batch_size])),
-                return_tensors='pt',
-                padding=True,
-                truncation=True,
-            )
-            input_ids = encoded_inputs['input_ids'].to(self.device)
-            attention_mask = encoded_inputs['attention_mask'].to(self.device)
-            outputs = self.model(
-                input_ids=input_ids,
-                attention_mask=attention_mask,
-            )
-
-            predicted_spans = self.tokenizer.decode_best_spans(
-                encoded_inputs,
-                outputs,
-                self.num_spans,
-                self.max_answer_length,
-                self.num_spans_per_passage,
-            )
-
-            for idx, span in enumerate(predicted_spans):
-                answers.append(
-                    Answer(
-                        text=span.text,
-                        score=float(span.span_score.cpu().detach().numpy()),
-                        ctx_score=float(span.relevance_score.cpu().detach().numpy()),
-                    )
+        answers = {}
+        topk_answers = []
+        prev_k = 0
+
+        for k in self.topk_em:
```

Comment:
We should try to make evaluation isolated from prediction. I.e. if we want to do interactive QA, gives an argument called related to EM is not making sense.

I would suggest, instead of put `self.topk_em` in the constructor, we should make it as an argument of `predict`
and instead of call it `topk_em` we name it as `milestones` (maybe not a perfect name, but something like this). Since here, it is regardless of `em` just where we generate answers

Reply:
And by default, i think the `milestone` list should be `[len(texts)]`?